### PR TITLE
ConfirmedWallet State

### PR DESCRIPTION
**Pull request by Sajma27**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [ ] Change is small and easy to review (please split big changes into multiple PRs)
- [ ] Change is consistent with architecture
- [ ] Change is consistent with test architecture
- [ ] Change is consistent with naming conventions
- [ ] New code is covered with tests
- [ ] Tests related to old code are updated
- [ ] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-sdk/src/api/wallet/ConfirmedWallet.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    readonly code: string,
+  ) {}
+
+  get asSerializableRequestedWallet(): SerializableConfirmedWallet {
+    return {
+      email: this.email,
+      ensName: this.ensName,
+      code: this.code,
+    };
+  }
```
Comment (xitronix) [inline universal-login-sdk/src/api/wallet/ConfirmedWallet.ts:16]: introduce `encryptWallet` fn which will call RelayerApi.encryptWallet

File: universal-login-sdk/src/core/models/WalletService.ts [hunk 2]
```diff
@@ -26,6 +31,10 @@ export type WalletState = {
 
 export type SerializedWalletState = {
   kind: 'None';
+} | {
+  kind: 'Confirmed';
+  name: string;
```
Comment (xitronix) [inline universal-login-sdk/src/core/models/WalletService.ts:36]: what is the purpose of this variable?

Comment (Sajma27) [reply universal-login-sdk/src/core/models/WalletService.ts:36]: I wasn't sure if we need it

Comment (Sajma27) [reply universal-login-sdk/src/core/models/WalletService.ts:36]: Should I remove it?

Comment (xitronix) [reply universal-login-sdk/src/core/models/WalletService.ts:36]: i think yes. You have everything inside ConfirmedWallet. We will add it if we need it

Comment (Sajma27) [reply universal-login-sdk/src/core/models/WalletService.ts]: Done

File: universal-login-sdk/src/core/models/WalletService.ts [hunk 3]
```diff
@@ -8,6 +9,10 @@ export type SerializedDeployingWallet = ApplicationWallet & {deploymentHash: str
 
 export type WalletState = {
   kind: 'None';
+} | {
+  kind: 'Confirmed';
+  name: string;
```
Comment (xitronix) [inline universal-login-sdk/src/core/models/WalletService.ts:14]: name?

Comment (Sajma27) [reply universal-login-sdk/src/core/models/WalletService.ts]: Sorry, my bad, done

File: universal-login-sdk/src/core/services/WalletService.ts [hunk 4]
```diff
@@ -265,7 +265,7 @@ export class WalletService {
   }
 
   getContractAddress() {
-    ensure(this.state.kind !== 'None', InvalidWalletState, 'not None', this.state.kind);
+    ensure(this.state.kind !== 'None' && this.state.kind !== 'Confirmed', InvalidWalletState, 'not None or Confirmed', this.state.kind);
```
Comment (xitronix) [inline universal-login-sdk/src/core/services/WalletService.ts:268]: split to 2 ensure calls

Comment (Sajma27) [reply universal-login-sdk/src/core/services/WalletService.ts]: Done

File: universal-login-sdk/src/api/wallet/ConfirmedWallet.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import {SerializableConfirmedWallet} from '@unilogin/commons';
+
+export class ConfirmedWallet implements SerializableConfirmedWallet {
+  constructor(
+    readonly email: string,
+    readonly ensName: string,
+    readonly code: string,
+  ) {}
+
+  get asSerializableRequestedWallet(): SerializableConfirmedWallet {
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/src/api/wallet/ConfirmedWallet.ts:10]: `asSerializableConirmedWallet`

**Pull request merged**